protocol/ibd/selected_tip.go

```diff
@@ -0,0 +1,133 @@
+package ibd
+
+import (
+	"errors"
+	"github.com/kaspanet/kaspad/blockdag"
+	"github.com/kaspanet/kaspad/netadapter/router"
+	"github.com/kaspanet/kaspad/protocol/common"
+	peerpkg "github.com/kaspanet/kaspad/protocol/peer"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/kaspanet/kaspad/wire"
+	"time"
+)
+
+const minDurationToRequestSelectedTips = time.Minute
+
+func requestSelectedTipsIfRequired(dag *blockdag.BlockDAG) error {
+	if hasRecentlyReceivedBlock(dag) {
+		return nil
+	}
+	return requestSelectedTips(dag)
+}
+
+func hasRecentlyReceivedBlock(dag *blockdag.BlockDAG) bool {
+	return dag.Now().Sub(dag.SelectedTipHeader().Timestamp) > minDurationToRequestSelectedTips
+}
+
+func requestSelectedTips(dag *blockdag.BlockDAG) error {
+	for _, peer := range peerpkg.ReadyPeers() {
+		peer.RequestSelectedTipIfRequired()
+	}
+	return nil
+}
+
+// RequestSelectedTip waits for selected tip requests and handles them
+func RequestSelectedTip(incomingRoute *router.Route,
+	outgoingRoute *router.Route, peer *peerpkg.Peer, dag *blockdag.BlockDAG) error {
+	for {
+		peer.WaitForSelectedTipRequests()
+
+		// We run the flow inside a func so that the defer is called at its end
+		err := func() error {
+			defer peer.FinishRequestingSelectedTip()
+
+			shouldContinue, err := requestSelectedTip(outgoingRoute)
```

Comment:
I think requestSelectedTip and receiveSelectedTip can be combined to a single function, unless we ever want to call them separately, which I highly doubt

Reply:
We spoke about this and decided to leave it as is.